elastic/elastic-agent
# Add filebeat receiver to otel mode

## What does this PR do?

Adds filebeat receiver as an otel receiver.

## Why is it important?

This is the first beat receiver, and this will allow us to test the feasibility of running beats as otel receivers.


## Checklist




## Disruptive User Impact

should be 0 user impact of existing configurations

## How to test this PR locally

Use something like the following as `otel.yml`, and run `elastic-agent otel --config otel.yml`
This will run the beat receiver, which will produce one event.
The event should be written to the path you specify in the `file/no_rotation` section.

```
receivers:
  filebeatreceiver:
    filebeat:
      inputs:
        - type: benchmark
          enabled: true
          count: 1
          message: "The quick brown fox jumped over the lazy dog"
    output:
      otelconsumer:
    logging:
      level: info
      selectors:
        - '*'
    path:
      home: /tmp/<state dir>
exporters:
  file/no_rotation:
    path: /tmp/<output file>
service:
  pipelines:
    logs:
      receivers: [filebeatreceiver]
      exporters: [file/norotation]
```

## Related issues



## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
